code/game/objects/items/spraypaint.dm

```diff
@@ -0,0 +1,42 @@
+/obj/item/spraypaint
+	name = "spraypaint"
+	desc = "Aerosol paint sealed in a pressurized container."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "spraypaint"
+	matter = list(MATERIAL_PLASTIC = 2, MATERIAL_STEEL = 1)
+	price_tag = 10
+	var/uses = 30
+	w_class = ITEM_SIZE_TINY
+	throwforce = WEAPON_FORCE_HARMLESS
+
+/obj/item/spraypaint/afterattack(atom/target, mob/user as mob, proximity)
+	if(!proximity) return
+	if(istype(target,/turf))
+		var/graffititype = input("Choose what you'd like to paint") in list("Kot","Onestar","Doodle","Piss","Clown","Skull","Heart")
+		playsound(loc, 'sound/effects/interaction/graffiti.ogg', 100, 1)
+		if(do_after(user, 2 SECONDS, target))
+			to_chat(user, "<span class='notice'>You start tagging \the [target.name]!</span>")
+		else
+			to_chat(user, "<span class='notice'>You must stand still while tagging \the [target.name]!</span>")
+			return FALSE
+		switch(graffititype)
+			if("Kot")
+				new /obj/effect/decal/cleanable/graffiti_kot(target)
+			if("Onestar")
+				new /obj/effect/decal/cleanable/graffiti_onestar(target)
+			if("Doodle")
+				new /obj/effect/decal/cleanable/graffiti_doodle(target)
+			if("Piss")
+				new /obj/effect/decal/cleanable/graffiti_piss(target)
+			if("Clown")
+				new /obj/effect/decal/cleanable/graffiti_clown(target)
+			if("Skull")
+				new /obj/effect/decal/cleanable/graffiti_skull(target)
+			if("Heart")
+				new /obj/effect/decal/cleanable/graffiti_heart(target)
+		if(uses)
+			uses--
+		if(uses <= 0)
+			playsound(loc, 'sound/effects/interaction/graffiti_empty.ogg', 100, 1)
+			to_chat(user, "<span class='notice'>You emptied the can</span>")
+			qdel(src)
```

Comment:
Should we really qdel it? A beaten-up sprite that can be recycled for part of materials would be nice, could also work for as criminal evidence.

Reply:
dealt with, added a empty sprite, and changed the delete to change the icon state

Reply:
Now test it again, as I'm sure it's going to work empty.
A new description would work nice too.

Reply:
yes, i wanted to resolve it too fast, maybe something like
if(icon_state = "spraypaint_empty")
    return FALSE
    
    will do i hope

Reply:
You can also before actual spray painting check for remaining uses.
And only upon actual use check if the spraycan needs to be emptier

Reply:
fixed, now when clicking on the turf, it plays the empty can sound, tell  you its empty and doesnt start nor spawn a graffiti, in short it works.

also added a new name and desc for the empty spray can; so it doesnt read spray can with the same desc when emptied.